### PR TITLE
Use ump call for create new

### PR DESCRIPTION
**Pull request by jatgarg**
(https://dev.azure.com/fluidframework/internal/_boards/board/t/Team%20Vlad/Features/?workitem=237)

For more information about how to contribute to this repo, visit this [page](https://github.com/microsoft/FluidFramework/blob/main/CONTRIBUTING.md)

## Description
Use ump post call for create new if the size of the body is less than 80Kb which is the limit the server accepts for the post body. This saves the options call and also prevents access token to be included in query param. p99 is less than 75kb.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/odspUtils.ts [hunk 1]
```diff
@@ -340,3 +340,7 @@ export function createCacheSnapshotKey(odspResolvedUrl: IOdspResolvedUrl): ICach
     };
     return cacheEntry;
 }
+
+// 80KB is the max body size that we can put in ump post body for server to be able to accept it.
+// Keeping it 78KB to be a little cautious. As per the telemetry 99p is less than 78KB.
+export const maxUmpPostBodySize = 78000;
```
Comment (AndreiZe) [inline packages/drivers/odsp-driver/src/odspUtils.ts:346]: 78 kb = 79,872

File: packages/drivers/odsp-driver/src/createFile.ts [hunk 2]
```diff
@@ -195,19 +197,48 @@ export async function createNewFluidFileFromSummary(
...
+                } else {
+                    const parts = getUrlAndHeadersWithAuth(
+                        initialUrl, storageToken, forceAccessTokenViaAuthorizationHeader);
+                    url = parts.url;
+                    headers = {
+                        ...parts.headers,
+                        "Content-Type": "application/json",
+                    };
+                    postBody = snapshotBody;
+                }
```
Comment (AndreiZe) [inline packages/drivers/odsp-driver/src/createFile.ts:231]: I wonder if there is a chance to extract code that uses multi-part body into utility and use it across all places where we employ ump=1 optimization. Non-blocking

**Review by AndreiZe — APPROVED** [review]
LGTM

File: packages/drivers/odsp-driver/src/createFile.ts [hunk 3]
```diff
@@ -195,19 +197,48 @@ export async function createNewFluidFileFromSummary(
             logger,
             { eventName: "createNewFile" },
             async (event) => {
-                const { url, headers } = getUrlAndHeadersWithAuth(
-                    initialUrl, storageToken, forceAccessTokenViaAuthorizationHeader);
-                headers["Content-Type"] = "application/json";
+                const snapshotBody = JSON.stringify(containerSnapshot);
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/createFile.ts:200]: Should UMP call be conditional on size of payload? What if it's larger than 80K?

Comment (jatgarg) [reply packages/drivers/odsp-driver/src/createFile.ts:200]: I do have a size check below it :)

File: packages/drivers/odsp-driver/src/createFile.ts (hunk 2 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/createFile.ts:231]: Yes. I will follow up on this.

**Pull request merged**